### DISCUSSION TITLE
New Feature: SQL query playground in development mode

### DISCUSSION THREAD

**Discussion by sksar** [pocketbase/pocketbase#6173] [Polls]
###  A basic SQL console, similar to PhpMyAdmin, would be beneficial for development purposes only.
Just a simple place in the sidebar where SQL statements can be written and tested and seen as output below the console.

- This feature would enable rapid prototyping and feature development using SQL and hooks.
- SQL statements could be executed from PocketBase hooks, similar to stored procedures in MySQL.

#### Here is an example (This is Tableplus)

![Screenshot 2024-12-24 at 10 46 26](https://github.com/user-attachments/assets/c6ecf722-9eec-4251-bc6d-df5c60d6b026)

**Comment by pnmcosta** [comment]
Honestly, I just don't see a point to it, there are plently of fully featured apps out there to query/troubleshoot and performance tune sqlite itself, and other db's too. You can easily download a backup from the admin currently and run those tools on the db file itself.

Perhaps I'm more inclined to other features that would benefit PB users/devs directly that are not already wildly available 😅 Scaffolding the new Base Record Proxies comes to mind.
